guppylang/checker/core.py

```diff
@@ -162,41 +162,45 @@ def __getitem__(self, item: DefId | str) -> Definition:
                 return assert_never(x)
 
 
+K = TypeVar("K")
+V = TypeVar("V")
+
+
 @dataclass
-class Locals:
+class Locals(Generic[K, V]):
```

Comment:
Its not exactly `Mapping` since the types are a bit different: `Mapping.__contains__` takes an `object` as key instead of `K`, `Mapping.keys` returns an `ItemsView` instead of a `set` etc.